common/utils.go

```diff
@@ -18,3 +23,83 @@ func ErrorMessages(errors []error) string {
 	}
 	return strings.Join(result, "\n")
 }
+
+func WithNetNS(ns netns.NsHandle, work func() error) error {
+	errChan := make(chan error)
+
+	go func() {
+		runtime.LockOSThread()
+		defer runtime.UnlockOSThread()
+
+		oldNs, err := netns.Get()
+		if err == nil {
+			defer oldNs.Close()
+
+			err = netns.Set(ns)
+			if err == nil {
+				defer netns.Set(oldNs)
+
+				err = work()
+			}
+		}
+
+		errChan <- err
+	}()
+
+	return <-errChan
+}
+
+type WeaveInterface struct {
```

Comment:
It is meant to represent an interface (in the network sense), and will probably sprout extra things like MTU in a future PR. `Link`?